File: static_prod/js/reports/hr/appraisal_report.js

export default {
    setup: {
        model: "Appraisal Report",
        title: "Appraisal Report",
        report_type: "script",
        include_opening: true,
        include_closing: true,
        allow_print: true,
        allow_download_csv: true,
        allow_download_excel: true,
        allow_download_pdf: true
    },
    filters: [
     
        {
            id: "appraisee",
            fieldlabel: "Appraisee",
            fieldname: "appraisee",
            fieldtype: "link",
            model: "Employee",
            columns: 1,
            placeholder: "",
        },

    ],
    actions: [
       
       
    ],
    columns: [
        {
            column_title: "Appraisal Date",
            column_name: "appraisal_date",
            column_type: "text",
            width: 200,
        },

        {
            column_title: "Appraisal Quarter",
            column_name: "appraisal_quarter",
            column_type: "text",
            width: 200,
            
        },
        {
            column_title: "Appraiser",
            column_name: "appraiser_name",
            column_type: "text",
            width: 200,
            
        },
        {
            column_title: "Appraisee",
            column_name: "appraisee_name",
            column_type: "text",
            width: 200,
        },
        {
            column_title: "Department",
            column_name: "department",
            column_type: "text",
            width: 200,
            
        },
        {
            column_title: "Total Open Ended Questions",
            column_name: "total_open_ended_questions",
            column_type: "text",
            width: 200,
        },
        {
            column_title: "Total closed Ended Questions",
            column_name: "total_closed_ended_questions",
            column_type: "text",
            width: 200,
        },

        {
            column_title: "Total Score (Open Ended)",
            column_name: "total_open_score",
            column_type: "text",
            width: 200,
        },
        {
            column_title: "Total Score (Closed Ended)",
            column_name: "total_closed_score",
            column_type: "text",
            width: 200,
        },
        {
            column_title: "Total Overall Score",
            column_name: "overall_score",
            column_type: "text",
            width: 200,
        },
    ]
}

File: controllers/hooks/reports/hr/appraisal_report.py
from controllers.utils import Utils

utils = Utils()
pp =utils.pretty_print

def appraisal_report(dbms, object):
    utils.pretty_print ("")
    appraisal_data_result = []
    appraisal_info = dbms.get_list("Appraisal",)
    if appraisal_info.status == utils.ok:
        appraisal_data = appraisal_info.data.rows
        for app in appraisal_data:
        
            if app.status == "Draft":
                continue 
        
            appraisal_row = {
                "appraisee_name": app.appraisee_name,  
                "appraiser_name": app.appraiser_name, 
                "appraisal_date": app.appraisal_date,
                "department": app.department,
                "appraisal_quarter": app.appraisal_quarter,
                "total_open_score": app.total_open_score,
                "total_closed_score": app.total_closed_score,
                "total_open_ended_questions": app.total_open_ended_questions,
                "total_closed_ended_questions":  app.total_closed_ended_questions,
                "overall_score": app.overall_score,
            }
            appraisal_data_result.append(appraisal_row)

    return utils.respond(utils.ok, {'rows': appraisal_data_result})
